ALBS-1044: Add build flavor repos when scheduling test tasks

javihernandez opened the pull request.

anfimovdm commented on alws/test_scheduler.py:
```diff
@@ -22,6 +22,49 @@ def __init__(self, term_event: threading.Event,
 
+
+    @staticmethod
+    def __get_repos_for_test_task(task: models.TestTask):
```
please, clarify return type

anfimovdm commented on alws/test_scheduler.py:
```diff
@@ -22,6 +22,49 @@ def __init__(self, term_event: threading.Event,
+            if item.type == 'rpm'
+            and item.arch == task.env_arch
+        ]
+        repos.extend(flavor_repos)
```
can it be better, if we will iterate by new arrays and add them in `repos`?
like this
```python3
build_repositories = [...]
linked_build_repos = [...]
flavor_repos = [...]
for repo_arr in (
    build_repositories,
    linked_build_repos,
    flavor_repos,
):
    repos.extend(repo_arr)
```
it's up to you

anfimovdm commented:
also, could you please create unit test for this code?

javihernandez commented:
> also, could you please create unit test for this code?

Added unit tests for the newly introduced function and fixed review comments, ready again for another round of review
